### PR TITLE
Add text about optional layer dependencies

### PR DESCRIPTION
**Pull request by DunderRoffe**
Signed-off-by: Viktor Sjölind <viktor.sjolind@pelagicore.com>

### CODE REVIEW

File: docs/articles/baseplatform/creating-a-new-layer.rst [hunk 1]
```diff
@@ -182,3 +182,18 @@ won't need to modify all the recipes.
 
 .. [#bbmanual] http://www.yoctoproject.org/docs/latest/bitbake-user-manual/bitbake-user-manual.html
 .. [#templateconf] http://www.yoctoproject.org/docs/latest/dev-manual/dev-manual.html#creating-a-custom-template-configuration-directory
```
Comment (tobsan) [inline docs/articles/baseplatform/creating-a-new-layer.rst]: I think all the footnotes should go in the bottom of the page.

File: docs/articles/baseplatform/creating-a-new-layer.rst [hunk 2]
```diff
@@ -182,3 +182,18 @@ won't need to modify all the recipes.
...
+
+Optionally depending on other layers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In some cases a layer might have recipes optional dependencies on other layers. For instance, one
+might want to append on a recipe for a Qt application which will only be built for some qt specific
+image. Adding the recipe directly to the layer will cause bitbake to generate errors about dangling
+appends when building non-Qt images. To mitigate this, create a "layers" directory containing
```
Comment (tobsan) [inline docs/articles/baseplatform/creating-a-new-layer.rst]: Perhaps one could elaborate more when specifying non-qt image, to say that such an image wouldn't include the qt layers - hence creating dangling appends.

File: docs/articles/baseplatform/creating-a-new-layer.rst [hunk 3]
```diff
@@ -182,3 +182,18 @@ won't need to modify all the recipes.
...
+directories named after the layers that is required, "qt5" in the Qt application example. Then
```
Comment (tobsan) [inline docs/articles/baseplatform/creating-a-new-layer.rst]: Perhaps it is worth mentioning that the naming has to match what the layer calls itself for this to work, at least with the current code that does this in meta-bistro and meta-pelux.

**Review by tobsan — CHANGES_REQUESTED** [review]

File: docs/articles/baseplatform/creating-a-new-layer.rst (hunk 1 above)
Comment (DunderRoffe) [reply docs/articles/baseplatform/creating-a-new-layer.rst]: fixed

File: docs/articles/baseplatform/creating-a-new-layer.rst (hunk 2 above)
Comment (DunderRoffe) [reply docs/articles/baseplatform/creating-a-new-layer.rst]: fixed

File: docs/articles/baseplatform/creating-a-new-layer.rst (hunk 3 above)
Comment (DunderRoffe) [reply docs/articles/baseplatform/creating-a-new-layer.rst]: Is not this implied by
> files in under ``layers/<layer-dependency>/`` are only included in ``BBPATH``

Also a high level description would be easier to maintain

**Comment by DunderRoffe** [comment]
> @tobsan
> Perhaps it is worth mentioning that the naming has to match what the layer calls itself for this to work, > at least with the current code that does this in meta-bistro and meta-pelux.



This is what I was trying to imply by "files in under layers/\<layer-dependency\>/ are only included in BBPATH". Maybe it needs to be clearer? Keep in mind that a high level description would be easier to maintain

File: docs/articles/baseplatform/creating-a-new-layer.rst [hunk 4]
```diff
@@ -180,5 +180,20 @@ and
 This can save you some later pain if the code gets moved to a new server, because in this way you
 won't need to modify all the recipes.
 
+Optionally depending on other layers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In some cases a layer might have recipes optional dependencies on other layers. For instance, one
+might want to append on a recipe for a Qt application which will only be built for some qt specific
```
Comment (jonte) [inline docs/articles/baseplatform/creating-a-new-layer.rst]: "Qt specific"

File: docs/articles/baseplatform/creating-a-new-layer.rst [hunk 5]
```diff
@@ -180,5 +180,20 @@ and
...
+image. Adding the recipe directly to the layer will cause bitbake to generate errors about dangling
+appends when building non-Qt images, which would not include any Qt layers. To mitigate this, create
+a ``layers`` directory containing directories named after the layers that is required, ``qt5`` in the
+Qt application example. Then add all .bb and .bbappend files needed to that layer. Finally in
```
Comment (jonte) [inline docs/articles/baseplatform/creating-a-new-layer.rst]: .bb -> `.bb`
.bbappend -> `.bbappend`

File: docs/articles/baseplatform/creating-a-new-layer.rst [hunk 6]
```diff
@@ -180,5 +180,20 @@ and
...
+``layer.conf`` make sure that the ``.bb`` and ``.bbappend`` files in under ``layers/<layer-dependency>/``
+are only included in ``BBPATH`` if the dependency is resolved, i.e. the layer is presant. An example
+of this can be found in meta-bistro_.
```
Comment (jonte) [inline docs/articles/baseplatform/creating-a-new-layer.rst]: Maybe give a reference to the exact file where this example can be seen in meta-bistro?

**Review by jonte — APPROVED** [review]
Looks good, I added a few nitpicks.

**Review by tobsan — APPROVED** [review]
Looks clearer now somehow.

**Pull request merged**